### PR TITLE
Auto-review tests to enforce `@group` annotations

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Needs #6770 for tests to pass.
Code based on my private repo.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
Under deliberation at the other PR.

File: tests/system/AutoReview/FrameworkCodeTest.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    '',
+                    strlen($relativePath) - strlen(DIRECTORY_SEPARATOR . $file->getBasename())
+                );
+
+                return sprintf(
+                    'CodeIgniter\\%s%s%s',
+                    strtr($relativePath, DIRECTORY_SEPARATOR, '\\'),
+                    $relativePath === '' ? '' : '\\',
+                    $file->getBasename('.' . $file->getExtension())
+                );
```
Comment (MGatner) [inline tests/system/AutoReview/FrameworkCodeTest.php:120]: Are all of our test classes really namespaced to `CodeIgniter`?

Comment (paulbalandan) [reply tests/system/AutoReview/FrameworkCodeTest.php:120]: Yes.

**Comment by paulbalandan** [comment]
Not sure, but `DatabaseLive` and `CacheLive` groups are not read?
```console
> vendor/bin/phpunit --list-groups
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

Available test group(s):
 - AutoReview
 - Others
 - SeparateProcess
 - default
```

**Comment by MGatner** [comment]
> Not sure, but `DatabaseLive` and `CacheLive` groups are not read?

That's very odd! I will do a little digging.

**Comment by MGatner** [comment]
I got nothing; everything looks correct to me. I think we proceed with this and then troubleshoot some more if the groups aren't actually working.

**Review by MGatner — APPROVED** [review]

**Comment by paulbalandan** [comment]
I did some testing on the `testsuites` and it seems the changes altered the groups.

**Comment by kenjis** [comment]
```console
$ vendor/bin/phpunit --list-groups
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

Available test group(s):
 - AutoReview
 - Others
 - SeparateProcess
 - default
```

**Comment by paulbalandan** [comment]
Cannot merge this for now. Coverage decreased to 15.9%. 🤯

**Comment by paulbalandan** [comment]
I am so confused why the tests hang! 😤

**Comment by kenjis** [comment]
```console
$ vendor/bin/phpunit --list-groups
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

Available test group(s):
 - AutoReview
 - CacheLive
 - DatabaseLive
 - Others
 - SeparateProcess
```

File: phpunit.xml.dist [hunk 2]
```diff
@@ -7,27 +7,30 @@
          beStrictAboutTodoAnnotatedTests="true"
          cacheResultFile="build/.phpunit.cache/test-results"
          colors="true"
+         columns="max"
```
Comment (MGatner) [inline phpunit.xml.dist:10]: Does this play nicely with GitHub Actions?

**Review by MGatner — APPROVED** [review]

File: phpunit.xml.dist (hunk 2 above)
Comment (paulbalandan) [reply phpunit.xml.dist:10]: It seems in GitHub Actions the display is already at max columns. Not sure how terminal width is computed there.

**Review by kenjis — APPROVED** [review]
I haven't looked into the details of the check implementation.

**Pull request merged**